hooks/useClickOutside.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (
+  // eslint-disable-next-line no-unused-vars
+  event: MouseEvent | TouchEvent
+) => void;
+
+export const useClickOutside = <T extends HTMLElement>(
+  handler: useClickOutsideType
+) => {
+  const ref = useRef<T>(null);
+  useEffect(() => {
+    const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'click';
```

Comment:
Just curious, Wouldn't just the `click` event is sufficient enough to make the hook work? 🤔

Reply:
if I have good memory it's to hollow usage of mouse and tactile device

Reply:
just click will be ok ?

Reply:
i think so, can you try it 🤔

Reply:
I will

Reply:
It's working !